Automattic/wp-calypso
# DIFM: Cleanup feature flag

#### Changes proposed in this Pull Request

* Removes usages of the `signup/redesigned-difm-flow` feature flag.
* Removes the `site-info-collection` signup step and associated state slice.

#### Testing instructions



* Navigate to `/start/do-it-for-me`.
* Go through the flow steps and complete checkout.
* Submit content on the Website Content step.
* Confirm that there are no errors during this flow.
